app/models/ems_refresh/save_inventory.rb

```diff
@@ -181,6 +181,26 @@ def save_vms_inventory(ems, hashes, target = nil)
     end
   end
 
+  # Convert all mapped hashes into actual tags and associate them with the object.
+  # The collection or collection[:tags] object should be an array of hashes, probably
+  # created by the ContainerLabelTagMapping.map_labels method. Each hash in the array
+  # should have the following basic structure:
+  #
+  # {:category_tag_id=>139, :entry_name=>"foo", :entry_description=>"bar"}
+  #
+  # The +collection+ argument can either be a Hash, in which case the argument
+  # should have a single :tags key, or a simple Array of hashes.
```

Comment:
I'm a little surprised you need to deal with {:tags => ...} form, IIUC https://github.com/ManageIQ/manageiq-providers-amazon/pull/183 creates :labels and :tags on same level, you have :labels and :tags in child keys on same level, and `save_child_inventory` calls `save_#{k}_inventory` on `hashes[k]` so I'd expect `save_tags_inventory` to not see the :tags, just the value.

But I don't mind.